main.tf

```diff
@@ -1,806 +1,548 @@
+      on_deregistration = target_failover.value.on_deregistration
+      on_unhealthy      = target_failover.value.on_unhealthy
+    }
+  }
 
-      dynamic "redirect" {
-        for_each = length(keys(lookup(default_action.value, "redirect", {}))) == 0 ? [] : [lookup(default_action.value, "redirect", {})]
+  target_type = try(each.value.target_type, null)
+  vpc_id      = try(each.value.vpc_id, var.vpc_id)
 
-        content {
-          path        = lookup(redirect.value, "path", null)
-          host        = lookup(redirect.value, "host", null)
-          port        = lookup(redirect.value, "port", null)
-          protocol    = lookup(redirect.value, "protocol", null)
-          query       = lookup(redirect.value, "query", null)
-          status_code = redirect.value["status_code"]
-        }
-      }
+  tags = merge(var.tags, try(each.value.tags, {}))
 
-      dynamic "fixed_response" {
-        for_each = length(keys(lookup(default_action.value, "fixed_response", {}))) == 0 ? [] : [lookup(default_action.value, "fixed_response", {})]
+  lifecycle {
+    create_before_destroy = true
+  }
+}
 
-        content {
-          content_type = fixed_response.value["content_type"]
-          message_body = lookup(fixed_response.value, "message_body", null)
-          status_code  = lookup(fixed_response.value, "status_code", null)
-        }
-      }
+################################################################################
+# Target Group Attachment
+################################################################################
 
-      # Authentication actions only available with HTTPS listeners
-      dynamic "authenticate_cognito" {
-        for_each = length(keys(lookup(default_action.value, "authenticate_cognito", {}))) == 0 ? [] : [lookup(default_action.value, "authenticate_cognito", {})]
+resource "aws_lb_target_group_attachment" "this" {
+  for_each = { for k, v in var.target_groups : k => v if local.create }
 
-        content {
-          # Max 10 extra params
-          authentication_request_extra_params = lookup(authenticate_cognito.value, "authentication_request_extra_params", null)
-          on_unauthenticated_request          = lookup(authenticate_cognito.value, "on_authenticated_request", null)
-          scope                               = lookup(authenticate_cognito.value, "scope", null)
-          session_cookie_name                 = lookup(authenticate_cognito.value, "session_cookie_name", null)
-          session_timeout                     = lookup(authenticate_cognito.value, "session_timeout", null)
-          user_pool_arn                       = authenticate_cognito.value["user_pool_arn"]
-          user_pool_client_id                 = authenticate_cognito.value["user_pool_client_id"]
-          user_pool_domain                    = authenticate_cognito.value["user_pool_domain"]
-        }
-      }
+  target_group_arn  = aws_lb_target_group.this[each.key].arn
+  target_id         = each.value.target_id
```

Comment:
When target group is used by ECS, target_type is ip but there is no target_id, this line fails. I think you should only create a `aws_lb_target_group_attachment` when there is a `each.value.target_id`, not sure about the right way to do that. I was able to deploy by changing the condition in the `for_each`:

```tf
  for_each = { for k, v in var.target_groups : k => v if false }
```

Reply:
If I understand correctly, you want to create a target group but not specify what is attached to that target group?

Reply:
Correct
It's very similar to this example with ALB and ECS on `terraform-aws-ecs`
https://github.com/terraform-aws-modules/terraform-aws-ecs/blob/master/examples/fargate/main.tf
![image](https://github.com/terraform-aws-modules/terraform-aws-alb/assets/1325474/99141442-8551-4fc1-bed2-f03f0916ed6d)

The attachment is managed by the autoscaling, so I guess we dont want to specify any ID in terraform

Reply:
this should be resolved in https://github.com/terraform-aws-modules/terraform-aws-alb/pull/305/commits/d112a97df83035ad9d5cbeda9139b554dba8b740

Reply:
It works, thanks a lot!

Reply:
Hi @bryantbiggs , sorry to reopen but it seems https://github.com/terraform-aws-modules/terraform-aws-alb/commit/d112a97df83035ad9d5cbeda9139b554dba8b740 has disappeared in the final MR
Should I open a new issue ?

Reply:
the functionality is still there, just different due to stability reasons - see https://github.com/terraform-aws-modules/terraform-aws-ecs/blob/736a7da9d4a5a514b1196c9b068050b11277b9a7/examples/fargate/main.tf#L237

Reply:
Oh my mistake thanks!